### PR TITLE
#1103: Improve content block GET API endpoint

### LINKED ISSUE #1103: Improve content block GET API endpoint

### PR DESCRIPTION
**Pull request by tim-bezhashvyly**
Closes #1103.

### CODE REVIEW

File: src/ContentBlock/ContentDelivery/ContentBlockService.php [hunk 1]
```diff
@@ -28,33 +22,21 @@ class ContentBlockService
...
     {
         try {
-            $snippetKeyGenerator = $this->snippetKeyGeneratorLocator->getKeyGeneratorForSnippetCode('content_block_'.$contentBlockName);
+            $snippetKeyGenerator = $this->snippetKeyGeneratorLocator->getKeyGeneratorForSnippetCode($contentBlockName);
             $key = $snippetKeyGenerator->getKeyForContext($context, []);
 
             return $this->dataPoolReader->getSnippet($key);
         } catch (KeyNotFoundException $e) {
-            throw new ContentBlockNotFoundException("Content block $contentBlockName does not exist");
+            throw new ContentBlockNotFoundException(sprintf('Content block "%s" does not exist.', $contentBlockName));
```
Comment (Schrank) [inline src/ContentBlock/ContentDelivery/ContentBlockService.php]: We pass the name from request directly back to the error if not existing. On normal pages one would call this XSS - not sure what to do in APIs.

https://twitter.com/Fabian_ikono/status/975634972550844416

File: tests/Integration/Suites/ContentBlockApiTest.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        $httpHeaders = HttpHeaders::fromArray([
+            'Accept' => 'application/vnd.lizards-and-pumpkins.content_blocks.v2+json',
+        ]);
+
+        $httpRequestBody = new HttpRequestBody(json_encode([
+            'content' => $contentBlockContent,
+            'context' => ['website' => 'fr', 'locale' => 'fr_FR'],
+            'data_version' => '-1',
+        ]));
+
```
Comment (Schrank) [inline tests/Integration/Suites/ContentBlockApiTest.php]: we can test for 404 before the put

File: src/ContentBlock/ContentDelivery/ContentBlockService.php (hunk 1 above)
Comment (Schrank) [reply src/ContentBlock/ContentDelivery/ContentBlockService.php]: Do we return json? We should always return json, then this is not our problem anymore imho

Comment (Schrank) [reply src/ContentBlock/ContentDelivery/ContentBlockService.php]: https://twitter.com/pete_cags/status/975642830134792192

File: tests/Integration/Suites/ContentBlockApiTest.php (hunk 2 above)
Comment (tim-bezhashvyly) [reply tests/Integration/Suites/ContentBlockApiTest.php]: Good idea.

Comment (tim-bezhashvyly) [reply tests/Integration/Suites/ContentBlockApiTest.php]: Implemented.

**Pull request merged**